backends/backends.c

```diff
@@ -1101,10 +1101,9 @@ void *backends_main(void *ptr) {
 #endif
 
 #if ENABLE_PROMETHEUS_REMOTE_WRITE
-    if(do_prometheus_remote_write) {
-        buffer_free(http_request_header);
+    buffer_free(http_request_header);
```

Comment:
We do:
`http_request_header = buffer_create(1);`
and:
`buffer_free(http_request_header)`
even when we don't use it at all. Shouldn't we put those 2 under `if(do_prometheus_remote_write)` as well?